### PR TITLE
Make the project function with .Net Framework classes again

### LINKED ISSUE #272: Version 2.5.0
I upgraded my projects to v2.5.0, and now CI is failing on my .NET Framework 4.7.1 tests (they still pass under .net Core 2.1).

Could not load file or assembly 'System.Runtime.Extensions, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies.

Going to have to downgrade for now.

### LINKED ISSUE #274: Error in Coverlet.Console 1.4.0
I just started getting this on my build server. Seems like you just pushed an update.

```
fake : An exception occurred while invoking executor 'executor://xunit/VsTestRunner2/net': The type initializer for 
'Coverlet.Core.Instrumentation.Tracker.xunit.runner.visualstudio.testadapter_72850aea-b6e0-4804-a443-564a9e718190' threw an exception.
At line:1 char:1
+ fake run Build.fsx
+ ~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (An exception oc...w an exception.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
Could not l
oad file or assembly 'System.Runtime.Extensions, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
Test Run Failed.
```

### LINKED ISSUE #285: Coverlet 2.5.0 throws Exception
I have updated my Coverlet.msbuild Version to 2.5.0. I am getting

> An exception occurred while invoking executor 'executor://xunit/VsTestRunner2/desktop_and_uap': The type initializer for 'Coverlet.Core.Instrumentation.Tracker.xunit.runner.visualstudio.testadapter_c
> -4aa5-a24d-de2d8256b22e' threw an exception.
> Could not load file or assembly 'System.Runtime.Extensions, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.

Exception. 

Project Information:
.net framework 4.6.1.

Also, in Version 2.3.0, It was failing my some tests due to rebuilding the project when enabled code coverage.

### PR DESCRIPTION
**Pull request by glennawatson**
This fixes #285, fixes #274 and fixes #272

Changed back to using a .NET Standard project for the coverlet.core. Allows for more targets (eg Xamarin based targets too potentially). 

Also do some redirecting of StdError for the process output, xunit for example sends errors there, and put them into the log.

.NET STandard 2.1 adds RelativePath() so using a place holder until that comes out.

### CODE REVIEW

File: src/coverlet.core/Coverage.cs [hunk 1]
```diff
@@ -300,5 +300,18 @@ private string GetSourceLinkUrl(Dictionary<string, string> sourceLinkDocuments,
             url = sourceLinkDocuments[keyWithBestMatch];
             return url.Replace("*", replacement);
         }
+
+        // This is coming to .NET Standard 2.1 as part of Path
+        private static string GetRelativePath(string relativeTo, string path)
```
Comment (tonerdo) [inline src/coverlet.core/Coverage.cs]: Move this method into a `PathHelper` class. See diff in this for an example https://github.com/tonerdo/coverlet/compare/remove-netcoreapp-dep?expand=1. That was me taking a stab at fixing this, feel free to ignore the implementation and just follow the structure

File: src/coverlet.core/coverlet.core.csproj [hunk 2]
```diff
@@ -1,15 +1,15 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
     <OutputType>Library</OutputType>
-    <TargetFramework>netcoreapp2.0</TargetFramework>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (tonerdo) [inline src/coverlet.core/coverlet.core.csproj]: Also switch `coverlet.msbuild.tasks.csproj` back to `netstandard2.0`

File: src/coverlet.core/Coverage.cs (hunk 1 above)
Comment (tonerdo) [reply src/coverlet.core/Coverage.cs]: Add unit tests as well

Comment (glennawatson) [reply src/coverlet.core/Coverage.cs]: I started to go through unit tests that the corefx team had and adapting them to my code 

https://github.com/dotnet/corefx/blob/a10890f4ffe0fadf090c922578ba0e606ebdd16c/src/System.Runtime.Extensions/tests/System/IO/Path.GetRelativePath.cs

I found that mine didn't handle unix based paths correctly.

The code you had also failed some of those unit tests.

I ended up creating a .netstandard template library which allows the injected template to work in .NET Framework classes.

**Comment by tonerdo** [comment]
@glennawatson I'm curious how your new approach fixes this issue

**Comment by glennawatson** [comment]
@tonerdo  The issue hasn't been that the injected instrumented class (because you've been hiding xunit's outut this has been obscured as part of other people's issues)

For example if I run against a .NET framework I get

```
System.TypeInitializationException : The type initializer for 'Coverlet.Core.Instrumentation.Tracker.Splat_2a9feed3-781f-482b-b285-66c3c2c3fa3d' threw an exception.
---- System.IO.FileNotFoundException : Could not load file or assembly 'System.Runtime.Extensions, Version=4.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
Stack Trace:
   at Coverlet.Core.Instrumentation.Tracker.Splat_2a9feed3-781f-482b-b285-66c3c2c3fa3d.RecordHit(Int32 )
   at Splat.TargetFrameworkExtensions.GetTargetFrameworkName(Assembly assembly)
   at Splat.Tests.ApiApprovalTests.CheckApproval(Assembly assembly, String memberName, String filePath) in C:\source\splat\src\Splat.Tests\API\ApiApprovalTests.cs:line 40
   at Splat.Tests.ApiApprovalTests.SplatProject() in C:\source\splat\src\Splat.Tests\API\ApiApprovalTests.cs:line 35
----- Inner Stack Trace -----
```

The fact that your msbuild task nor the application are .NET core has been the issue.

The new approach basically allows cecil to inject the dependencies based on .net standard rather than the .NET core ones.

**Comment by glennawatson** [comment]
Essentially since the new template project is .NET standard when you call `Instrumentor.AddCustomModuleTrackerToModule` it'll now when copying the field definitions and method definitions now be getting the .NET Standard versions of those copied values.

**Comment by glennawatson** [comment]
@tonerdo Any movement on this one? I can go back to a .net standard approach if you want and go through the work to make the PathHelper class work with sample paths, but this resolves all those issues.

**Pull request merged**